lib/termination-poller.js

```diff
@@ -0,0 +1,140 @@
+const assert = require('assert');
+const {runAWSRequest} = require('./aws-request');
+const Iterate = require('taskcluster-lib-iterate');
+const log = require('./log');
+
+const INSTANCE_LIMIT = 200;
+const POLL_DELAY = 100; // I forget which units this is :/
+
+class TerminationPoller {
+  constructor({ec2, regions, runaws = runAWSRequest, pollDelay = POLL_DELAY, state}) {
+    assert(typeof ec2 === 'object');
+    assert(Array.isArray(regions));
+    regions.forEach(region => assert(typeof region === 'string'));
+    assert(typeof state === 'object');
+
+    this.ec2 = ec2;
+    this.regions = regions;
+    this.state = state;
+    this.runaws = runaws;
+
+    this.iterator = new Iterate({
```

Comment:
Yep!  There's lots of great stuff we could be putting into lib-monitor in this patch which I haven't done yet.  I was planning to do a general once-over to check where we could/should be using lib-monitor as lots has changed since that was originally implemented.